dev/com.ibm.ws.kernel.boot.core/src/com/ibm/ws/kernel/provisioning/ServiceFingerprint.java

```diff
@@ -197,26 +176,21 @@ public static void put(File file) {
 
     private static void flush() {
         if (fingerPrintFile != null) {
-            FileWriter writer = null;
-            try {
-                File fingerPrintParent = fingerPrintFile.getParentFile();
-                if (fingerPrintParent != null && !!!fingerPrintParent.exists()) {
-                    //make the necessary directories, if the make fails issue a message
-                    if (!fingerPrintParent.mkdirs()) {
-                        System.out.println(MessageFormat.format(BootstrapConstants.messages.getString("warn.fingerprintUnableToMkDirs"),
-                                                                fingerPrintParent.getAbsolutePath(), fingerPrintFile.getAbsolutePath()));
-                    }
+            File fingerPrintParent = fingerPrintFile.getParentFile();
+            if (fingerPrintParent != null && !!!fingerPrintParent.exists()) {
+                //make the necessary directories, if the make fails issue a message
+                if (!fingerPrintParent.mkdirs()) {
+                    System.out.println(MessageFormat.format(BootstrapConstants.messages.getString("warn.fingerprintUnableToMkDirs"),
+                                                            fingerPrintParent.getAbsolutePath(), fingerPrintFile.getAbsolutePath()));
                 }
-                writer = new FileWriter(fingerPrintFile);
-                props.store(writer, null);
-            } catch (IOException e) {
-            } finally {
-                if (writer != null) {
-                    try {
-                        writer.close();
-                    } catch (IOException e) {
-                    }
+            }
+            try (DataOutputStream out = new DataOutputStream(new FileOutputStream(fingerPrintFile))) {
```

Comment:
props.store would use a BufferedWriter under the covers.  I would wrap the FileOutputStream in a BufferedOutputStream